helix-core/src/main/java/org/apache/helix/model/CloudConfig.java

```diff
@@ -238,14 +237,19 @@ public String getCloudInfoParserClassName () {
 
 
     private void validate() {
-      if (_clusterName == null) {
-        throw new HelixException("The cluster name/ID cannot be null.");
-      }
-
       if (_cloudEnabled) {
-        if (_cloudID == null || _cloudInfoURLList == null || _cloudInfoParserClassName == null || _cloudInfoURLList.isEmpty()) {
+        if (_cloudID == null ) {
+          throw new HelixException(
+              "This Cloud Configuration is Invalid. The CloudID is missing from the config.");
+        } else if ( _cloudInfoURLList == null) {
+          throw new HelixException(
+              "This Cloud Configuration is Invalid. The CloudInfoURLList is missing from the config.");
+        } else if (_cloudInfoParserClassName == null) {
+          throw new HelixException(
+              "This Cloud Configuration is Invalid. The CloudInfoParserClassName is missing from the config.");
+        } else if (_cloudInfoURLList.isEmpty()) {
```

Comment:
Combine this check with the previous cloudInfoURLList check to make it less confusing.

Reply:
Good suggestion. Fixed.